### PR TITLE
Fix CI tests

### PR DESCRIPTION
**Pull request by snewcomer**
Trying to get these tests passing. Fixes for ember-release and beyond

https://github.com/emberjs/ember.js/pull/19728

### CODE REVIEW

File: tests/unit/setup-context-test.js [hunk 1]
```diff
@@ -849,10 +849,7 @@ module('setupContext', function (hooks) {
 
     hooks.beforeEach(function () {
       const AppConfig = assign({ autoboot: false }, config.APP);
-      // .extend() because initializers are stored in the constructor, and we
-      // don't want to pollute other tests using an application created from the
-      // same constructor.
-      isolatedApp = App.extend({}).create(AppConfig);
```
Comment (rwjblue) [inline tests/unit/setup-context-test.js:855]: The thing this comment is describing is absolutely an issue (and why you had to add that `delete` in the after each), we can't share the same parent as `App` here. 

Can you bring back the comment, and assuming the goal was to get rid the manual `.extend()` call we can change to:

```js
isolatedApp = (class extends App {}).create(AppConfig);
```

This does the same thing as the `.extend` did and doesn't require the `delete` you added to the `afterEach`

File: tests/unit/setup-context-test.js [hunk 2]
```diff
@@ -862,6 +859,11 @@ module('setupContext', function (hooks) {
       setApplication(isolatedApp);
     });
 
+    hooks.afterEach(function () {
+      delete isolatedApp.constructor.initializers.foo;
+      isolatedApp.destroy();
```
Comment (rwjblue) [inline tests/unit/setup-context-test.js:864]: we should probably await settled after this

File: tests/unit/setup-context-test.js [hunk 3]
```diff
@@ -862,6 +859,11 @@ module('setupContext', function (hooks) {
       setApplication(isolatedApp);
     });
 
+    hooks.afterEach(function () {
+      delete isolatedApp.constructor.initializers.foo;
```
Comment (rwjblue) [inline tests/unit/setup-context-test.js:863]: let's remove this (in favor of adding the `class extends App {}` thing I mentioned in the other comment)

**Review by rwjblue — CHANGES_REQUESTED** [review]

File: tests/unit/setup-context-test.js (hunk 1 above)
Comment (snewcomer) [reply tests/unit/setup-context-test.js]: Thanks and addressed!

**Review by rwjblue — APPROVED** [review]

**Pull request merged**